src/WebJobs.Script.WebHost/Diagnostics/LinuxAppServiceEventGenerator.cs

```diff
@@ -110,7 +115,8 @@ private void WriteEvent(string eventPayload)
 
         public override void LogAzureMonitorDiagnosticLogEvent(LogLevel level, string resourceId, string operationName, string category, string regionName, string properties)
         {
-            _writeEvent($"{ScriptConstants.LinuxAzureMonitorEventStreamName} {(int)ToEventLevel(level)},{resourceId},{operationName},{category},{regionName},{NormalizeString(properties.Replace("'", string.Empty))},{DateTime.UtcNow}");
+            var timeStr = _azureMonitorLoggingOptions.Value.IsAzureMonitorTimeIsoFormatEnabled ? DateTime.UtcNow.ToString("s") : DateTime.UtcNow.ToString();
```

Comment:
Something to consider: with options objects I like to make them smart. Instead of simply holding config value, you can also add methods to them to _act_ on the config value. In this case you could add a `AzureMonitorLoggingOptions.GetUtcDateTime()` and it will return the date time with the correct format based on the config values.